api/src/router.rs

```diff
@@ -1,41 +1,43 @@
+use crate::endpoints::{book, identity, root};
 use crate::middleware;
+
 use warp::{filters::BoxedFilter, Filter, Reply};
 
 pub fn router() -> BoxedFilter<(impl Reply,)> {
-    warp::path::end()
+    let root = warp::path::end()
+        // GET - /
         .and(warp::get())
-        .and_then(crate::endpoints::root::root)
-        .boxed()
-        .or(warp::path("identity").and(
-            warp::path("oauth")
-                .and(
+        .and_then(root::root);
```

Comment:
This is missing `.boxed()`.

Reply:
I tried it with and without boxed and both worked and I couldn't measure a difference in compile time. So I think it is not needed for the intermediate filters, but only the very last one we return.

Reply:
Actually, adding to this and the `path!()` macro above. The macro does not add much, probably the difference is not really measurable, but the `.boxed()` had a big impact. If you have more endpoints the compile-time will go up to 5 minutes. That is what we struggled with during the whole last semester, until I figured out that you actually have to put more `.boxed()` in there.

Reply:
I actually can't measure a difference. I did some "benchmarks" with `cargo clean && time cargo build`, which resulted in following numbers:
* ad6034ea7c55919c65711accfd5a0ab1fc930ace
  * cargo build  349.11s user 21.90s system 663% cpu 55.893 total
  * cargo build  355.84s user 22.60s system 666% cpu 56.751 total
* ab272f73c51eee27b415e83405dd730b7fb85d73
  * cargo build  358.31s user 22.25s system 646% cpu 58.898 total
  * cargo build  357.41s user 22.87s system 649% cpu 58.575 total

I assume the performance difference is just for returned values, but don't make much of a difference for variables..